gallery/transforms/plot_torchscript_support.py

```diff
@@ -0,0 +1,134 @@
+"""
+===================
+Torchscript support
+===================
+
+.. note::
+    Try on `collab <https://colab.research.google.com/github/pytorch/vision/blob/gh-pages/main/_generated_ipynb_notebooks/plot_torchscript_support.ipynb>`_
+    or :ref:`go to the end <sphx_glr_download_auto_examples_transforms_plot_torchscript_support.py>` to download the full example code.
+
+This example illustrates `torchscript
+<https://pytorch.org/docs/stable/jit.html>`_ support of the torchvision
+:ref:`transforms <transforms>` on Tensor images.
+"""
+
+# %%
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+
+import torch
+import torch.nn as nn
+
+import torchvision.transforms as v1
+from torchvision.io import read_image
+
+plt.rcParams["savefig.bbox"] = 'tight'
+torch.manual_seed(1)
+
+# If you're trying to run that on collab, you can download the assets and the
+# helpers from https://github.com/pytorch/vision/tree/main/gallery/
+from helpers import plot
+ASSETS_PATH = Path('../assets')
+
+
+# %%
+# Most transforms support torchscript. For composing transforms, we use
+# :class:`torch.nn.Sequential` instead of
+# :class:`~torchvision.transforms.v2.Compose`:
+
+dog1 = read_image(str(ASSETS_PATH / 'dog1.jpg'))
+dog2 = read_image(str(ASSETS_PATH / 'dog2.jpg'))
+
+transforms = torch.nn.Sequential(
+    v1.RandomCrop(224),
+    v1.RandomHorizontalFlip(p=0.3),
+)
+
+scripted_transforms = torch.jit.script(transforms)
+
+plot([dog1, scripted_transforms(dog1), dog2, scripted_transforms(dog2)])
+
+
+# %%
+# .. warning::
+#
+#     Above we have used transforms from the ``torchvision.transforms``
+#     namespace, i.e. the "v1" transforms. The v2 transforms from the
+#     ``torchvision.transforms.v2`` namespace are the :ref:`recommended
+#     <v1_or_v2>` way to use transforms in your code.
+#
+#     The v2 transforms also support torchscript, but if you call
+#     ``torch.jit.script()`` on a v2 **class** transform, you'll actually end up
+#     with its (scripted) v1 equivalent.  This may lead to slightly different
+#     results between the scripted and eager executions due to implementation
+#     differences between v1 and v2.
+#
+#     If you really need torchscript support for the v2 transforms, **we
+#     recommend scripting the functionals** from the
+#     ``torchvision.transforms.v2.functional`` namespace to avoid surprises.
+#
+# Below we now show how to combine image transformations and a model forward
+# pass, while using ``torch.jit.script`` to obtain a single scripted module.
```

Comment:
This was a quick re-write and I didn't give this the same attention as the other examples. After 1+ week of writing docs I don't have much mental space left to dedicate to this. In particular, I didn't change anything to the text below. So if you have comments about it, and since everything below is pre-existing, I suggest to follow-up in other PRs if needed.